File: LLD/src/main/java/org/example/p1_oops/c5_interfaces/impl/Bird.java
package org.example.p1_oops.c5_interfaces.impl;

import org.example.p1_oops.c5_interfaces.contract.Flyable;

public class Bird implements Flyable {
    @Override
    public void fly() {
        System.out.println("Bird is flying in the sky.");
    }
}

File: LLD NodeJs/p1_oops/c5_interfaces/impl/Bird.js
const Flyable = require('../contract/Flyable');

/**
 * Bird class implementing Flyable
 * In JS, we extend the interface and implement the method
 */
class Bird extends Flyable {
    fly() {
        console.log("Bird is flying in the sky.");
    }
}

module.exports = Bird;
